src/main/resources/org/nest/sympy/misc.py

```diff
@@ -0,0 +1,178 @@
+from sympy.parsing.sympy_parser import parse_expr
+from sympy import *
+
+from sympy.matrices import zeros
+
+# Define constants
+MAX_ORDER = 10
+MAX_TRIES = 100
+
+# 't' is predefined in NESTML and represents time
+# 'derivative_factor' is the factor in 
+# shape'=derivative_factor*shape in case shape satisfies such an ODE.
+derivative_factor, t = symbols("derivative_factor, t")
+        
+
+class ShapeFunction(object):
+    '''The aim of this script is to check for any given function of
+    positive `t`, which we will call `shape`, if it satisfies a linear
+    homogeneous ODE. If it does, it will give it's order, initial
+    values, the equation itself and a list of all factors of
+    derivatives (which we will call `derivative_factors`) of the shape
+    function in the ODE, i.e. if shape''= a*shape+b*shape' the list
+    will be: [a,b].
+
+    Example:
+    ========
+
+    Convert the postsynaptic response shape from a function form  
+       shape = t * exp(t * a)
+    to an ODE form
+       shape'' = -a**2 * shape + 2*a * shape'
+    '''
+
+    def __init__(self, name, function):
+
+        self.name = parse_expr(name)
+
+        
+        # convert the shape function from a string to a symbolic expression
+        shape = parse_expr(function)
+
+        # found_ode is true if we find a linear homogeneous ODE that
+        # `shape` satisfies
+        found_ode = False
+        
+        # First we check if `shape` satisfies a linear homogeneous ODE
+        # of order 1. `derivatives` is a list of all derivatives of `shape`
+        # up to the order we are checking (which we just call 'order')
+        derivatives = [shape, diff(shape, t)]
+        
+        # If `diff_rhs_lhs`, here shape'-derivative_factors*shape=0 for
```

Comment:
The comment seems to be inseparable to me. I don't know how to explain the one without the other.